File: source-connector/src/main/java/sourceconnector/repository/file/validator/CompositeFileValidator.java
package sourceconnector.repository.file.validator;

import lombok.RequiredArgsConstructor;
import sourceconnector.repository.file.filter.FileFilter;

import java.util.List;

@RequiredArgsConstructor
public class CompositeFileValidator implements FileValidator {
  private final List<FileFilter> fileFilters;

  /**
   * If no file filter, always return {@code true}
   * @param filePath to validate
   */
  public boolean isValid(String filePath) {
    return this.fileFilters.stream()
      .allMatch(fileFilter -> fileFilter.accept(filePath));
  }
}

File: source-connector/src/main/kotlin/sourceconnector/repository/file/validator/CompositeFileValidator.kt
package sourceconnector.repository.file.validator

import offsetmanager.domain.file.FileKey
import sourceconnector.repository.file.filter.FileFilter

class CompositeFileValidator(fileFilters: List<FileFilter>) : FileValidator {
  private val fileFilters: List<FileFilter>

  init {
    require((fileFilters.isNotEmpty())) { "File filter condition cannot be null or empty" }
    this.fileFilters = fileFilters
  }

  /**
   * If no file filter, always return `true`
   * @param filePath to validate
   */
  override fun isValid(filePath: FileKey): Boolean {
    return fileFilters.all { fileFilter -> fileFilter.accept(filePath) }
  }
}
